database-plugins/src/main/java/co/cask/hydrator/plugin/DBUtils.java

```diff
@@ -58,6 +61,50 @@ public static void cleanup(Class<? extends Driver> driverClass) {
   }
 
   /**
+   * Ensures that the JDBC Driver specified in configuration is available and can be loaded. Also registers it with
+   * {@link DriverManager} if it is not already registered.
+   * @return {@link JDBCDriverShim} returns null if we are able to locate an existing driver for the
```

Comment:
Seems really strange to return `null` to indicate if we are able to locate and existing driver. Why is it so? Is it purely just for `DBUtils.deregisterShim(driverShim)` call? An API design like this make classes tightly coupled, which is not good. If the return `driverShim` is actually not used for normal operation but just for resource free up, it's better for this method to return a `Cancellable` or `Closeable` (non-null), which can do the resource cleanup inside (or no-op, depending on the result).